pkg/sql/create_table.go

```diff
@@ -1307,6 +1307,12 @@ func MakeTableDesc(
 		}
 	}
 
+	// If explicit primary keys are required, error out since a primary key was not supplied.
+	if len(desc.PrimaryIndex.ColumnNames) == 0 && desc.IsPhysicalTable() && evalCtx != nil &&
+		evalCtx.SessionData != nil && evalCtx.SessionData.RequireExplicitPrimaryKeys {
+		return desc, errors.Errorf("explicit primary key is required for table %s", desc.Name)
```

Comment:
I'd suggest tweaking this message to say why a primary key is required. Something like this:
```
no primary key specified for table %s (require_explicit_primary_keys = true)
```